src/pages/storage/StorageVolumes.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FC, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { MainTable, SearchBox, useNotify } from "@canonical/react-components";
+import Loader from "components/Loader";
+import { fetchStorageVolumes } from "api/storage-pools";
+import { isoTimeToString } from "util/helpers";
+import DeleteStorageVolumeBtn from "pages/storage/actions/DeleteStorageVolumeBtn";
+
+const StorageVolumes: FC = () => {
+  const notify = useNotify();
+  const [query, setQuery] = useState("");
+  const { name: pool, project } = useParams<{
+    name: string;
+    project: string;
+  }>();
+
+  if (!pool) {
+    return <>Missing storage pool name</>;
+  }
+  if (!project) {
+    return <>Missing project</>;
+  }
+
+  const {
+    data: volumes,
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.storage, pool, queryKeys.volumes],
+    queryFn: () => fetchStorageVolumes(pool),
+  });
+
+  if (error) {
+    notify.failure("Loading storage volumes failed", error);
+  }
+
+  if (isLoading) {
+    return <Loader text="Loading storage volumes..." />;
+  }
+  if (!volumes) {
+    return <>Loading storage volumes failed</>;
+  }
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Content type", sortKey: "contentType" },
+    { content: "Type", sortKey: "type" },
+    { content: "Created at", sortKey: "createdAt" },
+    { content: "Location", sortKey: "location" },
+    { content: "Config" },
+    { content: "Used by" },
+    {
+      content: "",
+      "aria-label": "Actions",
+    },
+  ];
+
+  const filteredVolumes = volumes.filter((volume) => {
+    return volume.name.toLowerCase().includes(query.toLowerCase());
+  });
+
+  const rows = filteredVolumes.map((volume) => {
+    return {
+      columns: [
+        {
+          content: volume.name,
+          role: "rowheader",
```

Comment:
This (and all the following) should be:
```suggestion
          role: "cell",
```